platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/network/connectivity/InboundConnectionHandlerTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.network.connectivity;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.common.test.fixtures.platform.TestPlatformContextBuilder;
+import com.swirlds.common.threading.interrupt.InterruptableConsumer;
+import com.swirlds.platform.Utilities;
+import com.swirlds.platform.crypto.KeysAndCerts;
+import com.swirlds.platform.network.Connection;
+import com.swirlds.platform.network.ConnectionTracker;
+import com.swirlds.platform.network.NetworkPeerIdentifier;
+import com.swirlds.platform.network.NetworkUtils;
+import com.swirlds.platform.network.PeerInfo;
+import com.swirlds.platform.system.address.AddressBook;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mockito;
+
+class InboundConnectionHandlerTest extends ConnectivityTestBase {
```

Comment:
When I run these tests locally, I notice that the number of certs used is exactly 1. This test should have a much larger set of peers... identifying a peer when there is only one peer doesn't really exercise this code the way it needs to be exercised.

Reply:
For the other tests that have a list of peers we should also ensure we have more than one peer (I did not check).

Reply:
we discussed this offline. Agreed to close.